### PR TITLE
feat: Add malicious contract interaction as an option

### PR DESCRIPTION
**Pull request by pnarayanaswamy**
Adds a button Malicious Contract Interaction to the PPOM section 

image

image

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1604,6 +1630,30 @@ const initializeFormElements = () => {
     console.log(result);
   };
 
+  // Malicious Mint
```
Comment (seaona) [inline src/index.js:1633]:
```suggestion
  // Malicious Contract Interaction
```

File: src/index.js [hunk 2]
```diff
@@ -1604,6 +1630,30 @@ const initializeFormElements = () => {
...
+      contractAddress = '0x00008F1149168C1D2fa1eBa1Ad3e9cD644510000';
+    }
+
+    const result = await provider.request({
+      method: 'eth_sendTransaction',
+      params: [
+        {
+          from: accounts[0],
+          to: contractAddress,
+          data: '0xef5cfb8c0000000000000000000000000b3e87a076ac4b0d1975f0f232444af6deb96c59',
```
Comment (seaona) [inline src/index.js:1649]: interesting! this calldata is calling the function `claimRewards` but that doesn't exist in the contracts I've checked. However I do see the Blockaid malicious warning, so I guess it's fine, even if that would make it revert

File: src/index.js [hunk 3]
```diff
@@ -1604,6 +1630,30 @@ const initializeFormElements = () => {
...
   };
 
+  // Malicious Mint
+  maliciousContractInteractionButton.onclick = async () => {
+    let contractAddress;
+
+    if (networkName === 'mainnet') {
+      contractAddress = '0x000062Accd1a9d62eF428eC86cA3dD4f45120000';
+    } else {
+      contractAddress = '0x00008F1149168C1D2fa1eBa1Ad3e9cD644510000';
```
Comment (seaona) [inline src/index.js:1640]: nit: I would rather move this into the [sample contracts file](https://github.com/MetaMask/test-dapp/blob/main/src/onchain-sample-contracts.js#L33), to make it easier to locate and maintain in the future (but not blocking)

Comment (pnarayanaswamy) [reply src/index.js]: done

File: src/index.js (hunk 1 above)
Comment (pnarayanaswamy) [reply src/index.js]: done

File: src/index.js (hunk 2 above)
Comment (pnarayanaswamy) [reply src/index.js:1646]: the payload and the accounts are what blockaid are using to test it, so I guess it is ok indeed

**Review by seaona — APPROVED** [review]
LGTM! nice addition :heart_eyes:

**Review by Blakerose8 — COMMENTED** [review]
Ok

**Review by Blakerose8 — COMMENTED** [review]
Ok

**Pull request merged**